### PR TITLE
Add countErrorMargin to WaitForControlledPods

### PR DESCRIPTION
**Pull request by jkaniuk**
Add countErrorMargin option to WaitForControlledPods

When using preemptibles on large scale, number of nodes do not become stable, so reaching DesiredPodCount could take a very long time.

### CODE REVIEW

**Comment by jkaniuk** [comment]
Tested manually, work as expected.

/cc @mm4tt 
/cc @wojtek-t

File: clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go [hunk 1]
```diff
@@ -78,6 +78,7 @@ type waitForControlledPodsRunningMeasurement struct {
 	kind                  string
 	selector              *measurementutil.ObjectSelector
 	operationTimeout      time.Duration
+	countErrorMargin      int
```
Comment (mm4tt) [inline clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go:81]: I know we haven't been doing it so far, but can we start documenting the fields here? They are, in a way, forming an external API (measurement api exposed in the yaml files) and deserve a sentence of documentation if meaning is not 100% clear.

**Review by mm4tt — COMMENTED** [review]
/lgtm
/approve

/hold

File: clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go (hunk 1 above)
Comment (jkaniuk) [reply clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go:85]: Done

Comment (mm4tt) [reply clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go:85]: Documentation in golang should be "a complete sentence that begins with the name of the element it describes" - https://blog.golang.org/godoc

**Review by mm4tt — COMMENTED** [review]
/lgtm

Thanks!

File: clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go [hunk 2]
```diff
@@ -482,13 +491,18 @@ func (w *waitForControlledPodsRunningMeasurement) waitForRuntimeObject(obj runti
 	o.lock.Lock()
 	defer o.lock.Unlock()
 	w.handlingGroup.Start(func() {
+		var minDesiredPodCount int
+		if w.countErrorMargin > 0 {
+			minDesiredPodCount = int(runtimeObjectReplicas) - w.countErrorMargin
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go:496]: I would do that slightly different, i.e. always set
minDesiredPodCount = runtimeObjectReplicas - w.countErrorMargin

[i.e always treat it as min] - I think the code would be a bit simpler below]

**Pull request merged**